### PR TITLE
Eval: Prefetching for heartbeat transactions

### PR DESCRIPTION
**Pull request by jannotti**
Add prefetching for the `HbAddress` of a Heartbeat transaction.

Where else might we need to notice that there's a new way for an account to appear in the new heartbeat txn type?

### CODE REVIEW

File: ledger/eval/prefetcher/prefetcher_alignment_test.go [hunk 1]
```diff
@@ -1422,3 +1423,54 @@ func TestEvaluatorPrefetcherAlignmentStateProof(t *testing.T) {
...
+				AccountBaseData: ledgercore.AccountBaseData{
+					MicroAlgos: basics.MicroAlgos{Raw: 1000001},
+				},
+			},
+			makeAddress(2): {
+				AccountBaseData: ledgercore.AccountBaseData{
+					MicroAlgos: basics.MicroAlgos{Raw: 100_000},
+				},
+				VotingData: basics.VotingData{
+					VoteID: otss.OneTimeSignatureVerifier,
```
[the remark this answers is not in the document]
Comment (jannotti) [reply ledger/eval/prefetcher/prefetcher_alignment_test.go:1452]: That is true right now. Let me think through whether it's correct.

First, heartbeat always uses the _current_ `VoteID`, not one from 320 rounds ago.  That seemed right to me, but I have to look more closely at how the `accountManager.Keys()` method works.  I need to ensure that if the service wants to heartbeat in round 10,000 if gets the current key, not the one for 9680.

If I keyreg and change my keys in round 10,000, in round 10,100 agreement will still be using my old keys, but I'm saying heartbeat should use new. Weird, but is that more or less weird than using my old keys in `apply.Heartbeat`?  Usually, we don't lookback 320 rounds in `apply`.

I'm beginning to lean toward changing to use the old keys. In that case, it's ok to add the VoteFirst and VoteLast checks, I think, though it should not matter.

[the remark this answers is not in the document]
Comment (jannotti) [reply ledger/eval/prefetcher/prefetcher_alignment_test.go:1452]: You're concerned that if we don't check VoteFirst/VoteLast that we could get an expired account keeping himself online while his keys are already expired?  I suppose that's true.  I think I'm leaning toward changing the keys that are used during `apply.Heartbeat` for more consistency, and adding the VoteFirst/VoteLast checks. They won't be a problem if we're using the old keys.

**Review by algorandskiy — APPROVED** [review]
The prefetcher update looks correct to me.

File: ledger/eval/prefetcher/prefetcher_alignment_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jannotti) [reply ledger/eval/prefetcher/prefetcher_alignment_test.go:1452]: I'd like to make that change in it's own PR, it'll probably take a while to get tests operating properly again.

**Pull request merged**